test-data/unit/check-parameter-specification.test

```
@@ -2144,7 +2165,7 @@ from typing_extensions import ParamSpec
 P= ParamSpec("P")
 T = TypeVar("T")
 
-def smoke_testable(*args: P.args, **kwargs: P.kwargs) -> Callable[[Callable[P, T]], Type[T]]:
+def smoke_testable(*args: P.args, **kwargs: P.kwargs) -> Callable[[Callable[P, T]], Type[T]]:  # E: ParamSpec "P" is unbound
```

this is functionality that mypy is losing, right? (you can see what it's going for / the OtherClass example is interesting)

i think that's fine (and seems fine looking at the new primer warnings), but maybe we should clarify this test case? e.g. the comments are out of date

---

Thanks! I've removed this test. This definition is not allowed by [PEP612](https://peps.python.org/pep-0612/#the-components-of-a-paramspec):

> These “properties” can only be used as the annotated types for *args and **kwargs, accessed from a ParamSpec already in scope.

*args and **kwargs do not add a ParamSpec to the scope, so the test was never conformant. IMO this is an extremely rare case: "given a set of arguments, produce a callable depending on those". I think dropping this implementation-defined extension to prevent silently accepting and misinterpreting invalid typing constructs is sensible.